app/models/icon.rb

```diff
@@ -22,7 +22,7 @@ class Icon < ApplicationRecord
   before_validation :use_icon_host
   before_save :use_https
   before_update :delete_from_s3
-  after_destroy :clear_icon_ids, :delete_from_s3
+  after_destroy_commit :clear_icon_ids, :delete_from_s3
```

Comment:
Does this still work with the destroyed? check and the key_was? Not sure what gets cleared in an after_commit vs after_destroy.

Reply:
I... kind of assumed we have tests on this that would be failing if it didn't? I think they're the same for this purpose, or at least rails guides didn't say differently.

Reply:
We have a job checking that it correctly queues `DeleteIconFromS3Job` on destroy: https://github.com/Marri/glowfic/blob/a8c6f790cb89ddea443c995cba8d562deb99a652/spec/models/icon_spec.rb#L102

Should be good to go!